qemu/tests/s390x_cpu_model_expansion.py

```diff
@@ -0,0 +1,33 @@
+
+
+def run(test, params, env):
+    """
+    Boot guest and query-cpu-model-expansion
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    vm = env.get_vm(params["main_vm"])
+    test.log.info('Start query cpu model supported by qmp')
+    # get cpu models for test
+    cpu_models = params.objects('cpu_models')
+    for cpu_model in cpu_models:
+        args = {'type': 'static', 'model': {'name': cpu_model}}
+        output = vm.monitor.cmd('query-cpu-model-expansion', args)
+        try:
+            model = output.get('model')
+            model_name = model.get('name')
+            model_props = model.get('props')
+            if model_name != cpu_model+'-base':
+                test.fail('Command query-cpu-model-expansion return'
+                          ' wrong model: %s with %s' % (cpu_model+'-base',
+                                                        model_name))
+            model_list = params.objects('cpu_models')
+            if model_name in model_list:
+                props = params['props']
```

Comment:
If possible, I would suggest using a dict and sorting by key to compare, because you can't guarantee that the format of the string will never change.

Reply:
solved